tests/test_sequences_and_iterators.py

```diff
@@ -179,7 +182,15 @@ def test_iterator_passthrough():
     """#181: iterator passthrough did not compile"""
     from pybind11_tests.sequences_and_iterators import iterator_passthrough
 
-    assert list(iterator_passthrough(iter([3, 5, 7, 9, 11, 13, 15]))) == [3, 5, 7, 9, 11, 13, 15]
+    assert list(iterator_passthrough(iter([3, 5, 7, 9, 11, 13, 15]))) == [
+        3,
+        5,
+        7,
+        9,
+        11,
+        13,
+        15,
```

Comment:
For cases like these, rewrite. The problem is that Python's syntax does not let you move the bracket to the next line, so there's really not a good way to do it. This would be the Black way:

```python
vals = [3, 5, 7, 9, 11, 13, 15]
assert list(iterator_passthrough(iter(vals))) == vals
```

I would argue this is far more readable than trying to build one massive line inside an assert statement!

Reply:
Black does the same thing with `with:` statements. But it's not a problem of Black, it's a problem with Python's syntax that is supposed to get fixed in 3.10.

Reply:
I'd argue that the line should be

```python
assert(list(iterator_passthrough(iter([3, 5, 7, 11, 13, 15]))) ==
                                      [3, 5, 7, 11, 13, 15])
```

so that you can immediately get a hint of what is intended to be equal, instead of parsing `list(foo(iter(vals)))`. I know that kind of formatting would never pass with Black.

Reply:
The fist example clearly says that the lists are the same by providing the same variable on both sides. Your example relies on the visual lists being lined up. What if you scale it to 100 items? What if you make a small mistake?

And, it does work fine with Black:

```python
# fmt: off
assert(list(iterator_passthrough(iter([3, 5, 7, 11, 13, 15]))) ==
                                      [3, 5, 7, 11, 13, 15])
# fmt: on
```

That's what fmt on/of are for. Telling Black _and_ the reader that there is a reason this is formatted this way. There's nothing wrong with using it in special cases, like matrices, etc. It's a signal "hey, the way I wrote this matters" (though only to the reader, Python can't detect anything from the formatting, and so can't tell you if you made a mistake in one of those two lists, for example).

Reply:
There are too many parenthesis already, by the way, to want to add another around the whole thing, IMO. But maybe that's just me. Yes, I like Ruby. :)